src/Features/Core/Portable/AddImport/AbstractAddImportFeatureService.cs

```diff
@@ -517,6 +517,74 @@ private static bool NotNull(SymbolReference reference)
             return fixesForDiagnosticBuilder.ToImmutableAndFree();
         }
 
+        public async Task<ImmutableArray<AddImportFixData>> GetUniqueFixesAsync(
+            Document document, TextSpan span, ImmutableArray<string> diagnosticIds,
+            ISymbolSearchService symbolSearchService, bool searchReferenceAssemblies,
+            ImmutableArray<PackageSource> packageSources, CancellationToken cancellationToken)
+        {
+            var client = await RemoteHostClient.TryGetClientAsync(document.Project, cancellationToken).ConfigureAwait(false);
+            if (client != null)
+            {
+                var result = await client.TryInvokeAsync<IRemoteMissingImportDiscoveryService, ImmutableArray<AddImportFixData>>(
+                    document.Project.Solution,
+                    (service, solutionInfo, callbackId, cancellationToken) =>
+                        service.GetUniqueFixesAsync(solutionInfo, callbackId, document.Id, span, diagnosticIds, searchReferenceAssemblies, packageSources, cancellationToken),
+                    callbackTarget: symbolSearchService,
+                    cancellationToken).ConfigureAwait(false);
+
+                return result.HasValue ? result.Value : ImmutableArray<AddImportFixData>.Empty;
+            }
+
+            return await GetUniqueFixesAsyncInCurrentProcessAsync(
+                document, span, diagnosticIds,
+                symbolSearchService, searchReferenceAssemblies,
+                packageSources, cancellationToken).ConfigureAwait(false);
+        }
+
+        private async Task<ImmutableArray<AddImportFixData>> GetUniqueFixesAsyncInCurrentProcessAsync(
+            Document document,
+            TextSpan span,
+            ImmutableArray<string> diagnosticIds,
+            ISymbolSearchService symbolSearchService,
+            bool searchReferenceAssemblies,
+            ImmutableArray<PackageSource> packageSources,
+            CancellationToken cancellationToken)
+        {
+            var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var placeSystemNamespaceFirst = documentOptions.GetOption(GenerationOptions.PlaceSystemNamespaceFirst);
+            var allowInHiddenRegions = document.CanAddImportsInHiddenRegions();
+
+            var semanticModel = await document.GetRequiredSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            var diagnostics = semanticModel.GetDiagnostics(span, cancellationToken)
```

Comment:
Moved comment from below
```suggestion

            // Get the diagnostics that indicate a missing import.
            var diagnostics = semanticModel.GetDiagnostics(span, cancellationToken)
```